frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/GraphTableDialog.java

```diff
@@ -0,0 +1,91 @@
+package com.virtuslab.gitmachete.frontend.actions.dialogs;
+
+import javax.swing.Action;
+import javax.swing.JComponent;
+
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.ui.ScrollPaneFactory;
+import com.intellij.ui.table.JBTable;
+import com.intellij.util.Consumer;
+import com.intellij.util.ui.JBDimension;
+import com.intellij.util.ui.JBUI;
+import org.checkerframework.checker.guieffect.qual.UI;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.binding.RuntimeBinding;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositorySnapshot;
+import com.virtuslab.gitmachete.frontend.ui.api.table.ISimpleGraphTableProvider;
+
+@UI
+public final class GraphTableDialog extends DialogWrapper {
+  private final JBTable table;
+  private final @Nullable IGitMacheteRepositorySnapshot repositorySnapshot;
+  private final int width;
+  private final int height;
+  private final @Nullable Consumer<IGitMacheteRepositorySnapshot> okAction;
+  private final boolean cancelButtonVisible;
+
+  public static GraphTableDialog of(IGitMacheteRepositorySnapshot gitMacheteRepositorySnapshot, String windowTitle,
+      @Nullable Consumer<IGitMacheteRepositorySnapshot> okAction, String okButtonText, boolean cancelButtonVisible) {
+    return of(gitMacheteRepositorySnapshot, windowTitle, /* width */ 800, /* height */ 500, okAction, okButtonText,
+        cancelButtonVisible);
+  }
+
+  public static GraphTableDialog of(IGitMacheteRepositorySnapshot gitMacheteRepositorySnapshot, String windowTitle, int width,
```

Comment:
For future usage to can easily set width and height

Reply:
Okay; pls use `window...` consistently (as in, now it's `windowTitle` but also just `width`)